app/src/main/kotlin/com/getstrm/pace/processing_platforms/synapse/SynapseViewGenerator.kt

```diff
@@ -114,30 +110,35 @@ class SynapseViewGenerator(
     }
 
     override fun toCondition(retention: DataPolicy.RuleSet.Filter.RetentionFilter): Condition {
-        val whereCondition = retention.conditionsList.headTailFold(
-            headOperation = { condition ->
-                DSL.`when`(
-                    toPrincipalCondition(condition.principalsList),
-                    condition.toSynapseRetentionCondition(),
-                )
-            },
-            bodyOperation = { conditionStep, condition ->
-                conditionStep.`when`(
-                    toPrincipalCondition(condition.principalsList),
-                    condition.toSynapseRetentionCondition(),
-                )
-            },
-            tailOperation = { conditionStep, condition ->
-                conditionStep.otherwise(condition.toSynapseRetentionCondition())
-            },
-        )
+        val retentionCondition = if (retention.conditionsList.size == 1) {
+            // If there is only one filter it should be the only option
+            retention.conditionsList.first().toSynapseRetentionCondition()
+        } else {
+            retention.conditionsList.headTailFold(
+                headOperation = { condition ->
+                    DSL.`when`(
+                        toPrincipalCondition(condition.principalsList),
+                        condition.toSynapseRetentionCondition(),
+                    )
+                },
+                bodyOperation = { conditionStep, condition ->
+                    conditionStep.`when`(
+                        toPrincipalCondition(condition.principalsList),
+                        condition.toSynapseRetentionCondition(),
+                    )
+                },
+                tailOperation = { conditionStep, condition ->
+                    conditionStep.otherwise(condition.toSynapseRetentionCondition())
+                },
+            )
+        }
 
         val retentionClause = DSL.field(
             "{0} > {1}",
             Boolean::class.java,
             DSL.timestampAdd(
                 DSL.field(DSL.unquotedName(retention.field.fullName()), Timestamp::class.java),
-                DSL.field("(${whereCondition})", Int::class.java),
+                DSL.field("(${retentionCondition})", Int::class.java),
```

Comment:
Unrelated to this PR, but why are you passing this as a string to `DSL.field`? Can't use the `retentionCondition`'s type?

Reply:
yes, fixed

Reply:
Reverted as I need the brackets around the retention condition